### PR TITLE
Validation & Testing

### PR DESCRIPTION
**Pull request by GeekyHacks**
In the following PR I did:
- Added the following validations:
       - For the User model:
               Name must not be blank.
               PostsCounter must be an integer greater than or equal to zero.
       - For the Post model:
               Title must not be blank.
               Title must not exceed 250 characters.
               CommentsCounter must be an integer greater than or equal to zero.
               LikesCounter must be an integer greater than or equal to zero.
- Added unit specs for all of your models' methods and validations.
- Fix linter errors

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
```
Comment (gemmen29) [inline spec/models/user_spec.rb:1]: - [ ] Kindly add tests for all custom methods in all models as per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec.md#:~:text=Add%20unit%20specs%20for%20all%20of%20your%20models%27%20methods%20and%20validations.).

**Review by gemmen29 — CHANGES_REQUESTED** [review]
Hi @GeekyHacks, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights 💯 
- No linters errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 

## Required Changes ♻️
_Check the comments under the review._



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/user_spec.rb (hunk 1 above)
Comment (GeekyHacks) [reply spec/models/user_spec.rb:1]: can you clarify please

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -1,41 +1,54 @@
...
+  context '#most_recent_posts' do
+    it 'should show last three posts for user' do
+      comment1 = Comment.create(user: @user, post: @post, text: 'Hello Abdu....')
+      comment2 = Comment.create(user: @user, post: @post, text: 'Hello Abdu....')
+      comment3 = Comment.create(user: @user, post: @post, text: 'Hello Abdu....')
+      comment4 = Comment.create(user: @user, post: @post, text: 'Hello Abdu....')
+      comment5 = Comment.create(user: @user, post: @post, text: 'Hello Abdu....')
+      comment6 = Comment.create(user: @user, post: @post, text: 'Hello Abdu....')
+
+      expected_posts = [comment1, comment2, comment3, comment4, comment5, comment6]
```
Comment (souad988) [inline spec/models/post_spec.rb]: Well done by testing this functionality but please make sur to use valid test and variable names as the task here is about comments not posts  #most_recent_posts , expected_posts  refers to posts not comments

**Review by souad988 — APPROVED** [review]
Hi @GeekyHacks , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Professional README
Good pr description
Add tests for main functionalities

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/post_spec.rb (hunk 2 above)
Comment (GeekyHacks) [reply spec/models/post_spec.rb]: Thank you

**Pull request merged**